contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -459,18 +468,49 @@ contract VaultAdmin is VaultStorage {
      */
     function _harvest(address _strategyAddr) internal {
         IStrategy strategy = IStrategy(_strategyAddr);
-        address rewardTokenAddress = strategy.rewardTokenAddress();
-        if (rewardTokenAddress != address(0)) {
-            strategy.collectRewardToken();
+        address[] memory rewardTokenAddresses = strategy.getRewardTokenAddresses();
+        bool collectReward = true;
+
+        for (uint256 i = 0; i < rewardTokenAddresses.length; i++) {
+            if (rewardTokenAddresses[i] == address(0)) {
```

Comment:
It would probably be better to swap this around so that if any reward token is set, we call collect

Reply:
Hmm yeah that makes much more sense... thx changing

Reply:
Looking at this code again, we will probably never have zero address elements in the array right? So something like this might make more sense:
```
if (rewardTokenAddresses.length > 0) {
  strategy.collectRewardTokens();
}
```